### PR TITLE
Excel export with details

### PR DESCRIPTION
**Pull request by zaucker**
- Add translated filter group outputs to Excel report
- Fix issue #330

### CODE REVIEW

File: lib/Agrammon/OutputFormatter/Excel.pm6 [hunk 1]
```diff
@@ -16,63 +18,77 @@ sub output-as-excel(
...
+    for ($output-sheet, $output-sheet-formatted) -> $sheet {
+        $sheet.set(0, 0, $dataset-name, :bold);
+        $sheet.columns[0] = Spreadsheet::XLSX::Worksheet::Column.new:
+                :custom-width, :width(50);
+        $sheet.columns[2] = Spreadsheet::XLSX::Worksheet::Column.new:
+                :custom-width, :width(32);
+    }
+
     my $row = 2;
+    my $lastModule = '';
```
Comment (jnthn) [inline lib/Agrammon/OutputFormatter/Excel.pm6:35]: Maybe good not to mix camel case and kebab case.

File: lib/Agrammon/OutputFormatter/Excel.pm6 [hunk 2]
```diff
@@ -16,63 +18,77 @@ sub output-as-excel(
...
                 my $var-print = $model.output-print($module, $output) ~ ',All';
                 if not $prints or $var-print.split(',') ∩ @print-set {
                     $n++;
                     my $unit = $model.output-unit($module, $output, $language);
-                    $output-sheet.set($row, $col+0, $module);
-                    $output-sheet.set($row, $col+2, $output);
-                    $output-sheet.set($row, $col+3, $val, :number-format('#,###'));
-                    $output-sheet.set($row, $col+4, $unit);
+                    my $outputLabel = $language ?? $model.output-labels($module, $output){$language} !! $output;
+                    my $unitLabel   = $language ?? $model.output-units($module, $output){$language}  !! $unit;
```
Comment (jnthn) [inline lib/Agrammon/OutputFormatter/Excel.pm6:47]: More camels

File: lib/Agrammon/OutputFormatter/Excel.pm6 [hunk 3]
```diff
@@ -16,63 +18,77 @@ sub output-as-excel(
...
                         if not $prints or $var-print.split(',') ∩ @print-set {
                             $n++;
                             my $unit = $model.output-unit($module, $output, $language);
-                            $output-sheet.set($row, $col+0, $module);
-                            $output-sheet.set($row, $col+1, $q-name);
-                            $output-sheet.set($row, $col+2, $output);
-                            $output-sheet.set($row, $col+3, $val, :number-format('#,###'));
-                            $output-sheet.set($row, $col+4, $unit);
+                            my $outputLabel = $language ?? $model.output-labels($module, $output){$language} !! $output;
+                            my $unitLabel   = $language ?? $model.output-units($module, $output){$language}  !! $unit;
```
Comment (jnthn) [inline lib/Agrammon/OutputFormatter/Excel.pm6:76]: And here

File: lib/Agrammon/OutputFormatter/Excel.pm6 [hunk 4]
```diff
@@ -84,6 +100,22 @@ sub output-as-excel(
...
 
+sub render-filters($sheet, $row is rw, $col, $model, Agrammon::Outputs::FilterGroupCollection $collection,
+                   $unit, $language, Bool :$all-filters) {
+    my @results = $collection.results-by-filter-group(:all($all-filters));
+    for @results {
+        my %keyFilters := .key;
+        my @filters = translate-filter-keys($model, %keyFilters).map: -> $trans { %( label => $trans.key, enum => $trans.value ) };
+        my $value := .value;
+        for @filters -> $filter {
+            $sheet.set($row, $col+2, $filter<enum>{$language}, :horizontal-align(RightAlign));
```
Comment (jnthn) [inline lib/Agrammon/OutputFormatter/Excel.pm6:111]: We don't need to build up the throw-away hash here; in the JSON one its there to get us over the JSON boundary. Here we can just do something like `my %filters := translate-filter-keys($model, %keyFilters);` and then `for %filters.kv -> %label, %enum { }`.

File: lib/Agrammon/OutputFormatter/Excel.pm6 (hunk 1 above)
Comment (zaucker) [reply lib/Agrammon/OutputFormatter/Excel.pm6:35]: Too much JS lately ... fixed.

File: lib/Agrammon/OutputFormatter/Excel.pm6 (hunk 2 above)
Comment (zaucker) [reply lib/Agrammon/OutputFormatter/Excel.pm6:47]: Fixed

File: lib/Agrammon/OutputFormatter/Excel.pm6 (hunk 3 above)
Comment (zaucker) [reply lib/Agrammon/OutputFormatter/Excel.pm6:76]: Fixed

File: lib/Agrammon/OutputFormatter/Excel.pm6 (hunk 4 above)
Comment (zaucker) [reply lib/Agrammon/OutputFormatter/Excel.pm6:111]: Fixed (left comment there on purpose).

**Review by jnthn — APPROVED** [review]

**Pull request merged**